src/Nfts/Nfts.tsx

```diff
@@ -4,61 +4,72 @@ import {RefreshControl, ScrollView, StyleSheet, Text, View} from 'react-native'
 import {SafeAreaView} from 'react-native-safe-area-context'
 
 import {Icon, Spacer} from '../components'
-import {useFilteredNfts} from './hooks'
+import {useSearch} from '../Search/SearchContext'
+import {useSelectedWallet} from '../SelectedWallet'
+import {useNfts, YoroiNft} from '../yoroi-wallets'
+import {filterNfts} from './filterNfts'
 import {ImageGallery, SkeletonGallery} from './ImageGallery'
 import {useNavigateTo} from './navigation'
 import {NoNftsScreen} from './NoNftsScreen'
 
 export const Nfts = () => {
   const navigateTo = useNavigateTo()
-  const handleNftSelect = (index: number) => navigateTo.nftDetails(nfts[index].id)
   const [isManualRefreshing, setIsManualRefreshing] = React.useState(false)
+  const wallet = useSelectedWallet()
   const strings = useStrings()
 
-  const {search, nfts, isLoading, refetch, isError} = useFilteredNfts({
+  const {isLoading, nfts, refetch, isError} = useNfts(wallet, {
     onSettled: () => {
       if (isManualRefreshing) setIsManualRefreshing(false)
     },
   })
 
+  const {search: nftsSearchTerm} = useSearch()
+  const filteredNfts = filterNfts(nftsSearchTerm, nfts)
+  const sortedNfts = filteredNfts.sort(sortNfts)
+  const nftsSearchResult = filterNfts(nftsSearchTerm, sortedNfts)
+
+  const hasEmptySearchResult = nftsSearchTerm.length > 0 && nftsSearchResult.length === 0
+  const hasNotNfts = nftsSearchTerm.length === 0 && nftsSearchResult.length === 0
```

Comment:
why do you include search term when the wallet doesn't have any nfts?

Reply:
if the wallet doesn't have any nfts, don't show search

Reply:
> if the wallet doesn't have any nfts, don't show search

 I think this is something to be considered by product